### PR TITLE
nixos/console: move enable option out of let in

### PR DESCRIPTION
**Pull request by SuperSandro2000**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/config/console.nix [hunk 1]
```diff
@@ -34,14 +34,17 @@ let
...
   };
-
-  setVconsole = !config.boot.isContainer;
 in
 
 {
   ###### interface
 
   options.console  = {
+    enable = mkEnableOption (lib.mdDoc "vconsole") // {
```
Comment (rnhmjoj) [inline nixos/modules/config/console.nix:43]: Some description of what this module does exactly wouldn't hurt.
Like "Enable configuration of the Linux virtual console".

Comment (SuperSandro2000) [reply nixos/modules/config/console.nix:43]: mkEnableOption will generate the following right now ``Whether to enable vconsole.
``

**Review by thiagokokada — APPROVED** [review]

File: nixos/modules/config/console.nix (hunk 1 above)
Comment (rnhmjoj) [reply nixos/modules/config/console.nix:43]: I know, and it's not very hepful.

Comment (SuperSandro2000) [reply nixos/modules/config/console.nix:43]: No, this will disable that systemd will setup virtual consoles so they are no longer available.

Comment (rnhmjoj) [reply nixos/modules/config/console.nix]: I know what it does, I wrote the module.
I'm telling you the description "enable vconsole" is next to useless.

File: nixos/modules/virtualisation/container-config.nix [hunk 2]
```diff
@@ -7,6 +7,9 @@ with lib;
   config = mkIf config.boot.isContainer {
 
     # Disable some features that are not useful in a container.
+
+    console.enable = false;
```
Comment (LunNova) [inline nixos/modules/virtualisation/container-config.nix:11]: maybe mkDefault like the other lines?

**Comment by LunNova** [comment]
Tried out `false`, does not disable the `getty` target/services.

**Comment by rnhmjoj** [comment]
> Tried out false, does not disable the getty target/services.

It's not mean to: this module solely takes care of configuring the console (keymap, font, colors, ...), not providing one.
That's why I say the "enable virtual console" description is misleading.

**Comment by LunNova** [comment]
Ok, that makes sense.

I think if this goes in it either needs a doc change like you asked for, or to also toggle the getty service so it does what the current docs imply.

File: nixos/modules/config/console.nix (hunk 1 above)
Comment (SuperSandro2000) [reply nixos/modules/config/console.nix]: > Like "Enable configuration of the Linux virtual console".

pushed that

**Comment by SuperSandro2000** [comment]
> to also toggle the getty service so it does what the current docs imply.

Added more getty units from https://github.com/nix-community/NixOS-WSL/blob/a4bbcf329a98a7faf79d10370278f87e836e1d7c/modules/wsl-distro.nix#L102-L105

If anything else should be improved please make a suggestion.

**Review by rnhmjoj — APPROVED** [review]

**Pull request merged**